util/uri/src/uri.rs

```diff
@@ -59,6 +59,12 @@ impl<Scheme: UriScheme> Uri<Scheme> {
             .expect("should never fail on valid url");
         self.port = port;
     }
+
+    /// Extract the subdomain from url
+    pub fn subdomain(&self) -> Option<&str> {
```

Comment:
I find this name somewhat confusing since it's really returning only the first component of a hostname. Since it's only used in two places I would just do it there, but separately why not use the entire hostname in those places? I think tying it to just the first part of the hostname is very specific to our deployment configuration.

Reply:
It's an attempt to limit the amount of superfluous data we send (and pay quite a bit for) to grafana cloud. It is indeed a bit specific to our deployment or k8s in-general however:

An example full host would be: `fog-services-fog-view-1-store-sfgdc-0-100000-lb.<namespace>.svc.cluster.local`. The `<namespace>` part is already getting sent as a seperate namespace label and the remainder `svc.cluser.local` is just static. So we really just need the subdomain to tell us which store is producing the error

Reply:
Gotcha. I think if we can avoid stuff that is super specific to our deployment that's generally good practice, but if it has tangible cost savings that might be a reason to still do that.